### PR TITLE
4248 push build version

### PR DESCRIPTION
**Pull request by pankaj-cgi**
This branch contains changes for:
-- Adding version information in manifest file
-- Adding a new version route to provide version information
-- Updates in existing Jenkins jobs to pass the values for manifest entries
-- Updates in Jenkins jobs to use single image repository on tools namespace in OpenShift

### CODE REVIEW

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/VersionRoute.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    
+	@Override
+    public void configure() {
+    	  	
+        from("jetty:http://{{hostname}}:{{port}}/version").routeId("hnsecure-version")
+        	.log("HNSecure received a request for version information")
+        	.to("log:HttpLogger?level=DEBUG&showBody=true&multiline=true").id("hnsecure-version-id")
+        	.setHeader(Exchange.CONTENT_TYPE, constant("application/json"))
+        	.process(new PopulateVersionInformation())
+        	.log("HNSecure sent a response with version infomration")
```
Comment (tschia) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/VersionRoute.java:19]: Typo in information

File: hnsecure/pom.xml [hunk 2]
```diff
@@ -217,6 +220,11 @@
               <transformers>
                 <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
                   <mainClass>ca.bc.gov.hlth.hnsecure.HnsEsbMainMethod</mainClass>
+                  <manifestEntries>
+					<Name>ca/bc/gov/hlth/hnsecure/</Name>
+					<Implementation-Version>${project.version}-${BRANCH_NAME}-${BUILD_NUMBER}</Implementation-Version>	
+                  	<Git-Commit>${git-commit}</Git-Commit>
```
Comment (weskubo-cgi) [inline hnsecure/pom.xml:226]: Isn't this just pulling the hardcoded value SNAPSHOT from the property git-commit?
Edit: I see how it's populated in the build. I'd be inclined to just default it to blank. SNAPSHOT doesn't mean much on it's own.

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/VersionRoute.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package ca.bc.gov.hlth.hnsecure.routes;
+
+import org.apache.camel.Exchange;
+
+import ca.bc.gov.hlth.hnsecure.parsing.PopulateVersionInformation;
+
+public class VersionRoute extends BaseRoute {
+	
+
```
Comment (weskubo-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/VersionRoute.java:9]: There's extra empty lines here and at the bottom of the file.

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/VersionRoute.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package ca.bc.gov.hlth.hnsecure.routes;
+
+import org.apache.camel.Exchange;
+
+import ca.bc.gov.hlth.hnsecure.parsing.PopulateVersionInformation;
+
+public class VersionRoute extends BaseRoute {
```
Comment (weskubo-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/routes/VersionRoute.java:7]: Maybe add a quick class comment explaining the purpose of this route.

File: hnsecure/src/test/java/ca/bc/gov/hlth/hnsecure/routes/VersionRouteTest.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+import org.apache.camel.Produce;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.builder.AdviceWithRouteBuilder;
+import org.apache.camel.spi.PropertiesComponent;
+import org.apache.camel.test.junit4.CamelTestSupport;
+import org.junit.Before;
+import org.junit.Test;
+import ca.bc.gov.hlth.hnsecure.properties.ApplicationProperties;
+
```
Comment (weskubo-cgi) [inline hnsecure/src/test/java/ca/bc/gov/hlth/hnsecure/routes/VersionRouteTest.java:11]: There's some extra empty lines in this class.

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateVersionInformation.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+import ca.bc.gov.hlth.hncommon.util.LoggingUtil;
+import ca.bc.gov.hlth.hnsecure.HnsEsbMainMethod;
+import net.minidev.json.JSONObject;
+
+/**
+ * This class is created to get the manifest information and provide the same in version route as JSON object.
+ * Manifest information is set in pom.xml 
+ */
+public class PopulateVersionInformation implements Processor{
```
Comment (weskubo-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateVersionInformation.java:16]: Missing space after Processor.

File: hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateVersionInformation.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+	 * Loading the version information using Package class.
+	 * Version information is set in jar's META-INF/Manifest.mf file
+	 * @return JSONObject
+	 */
+	protected static JSONObject getVersionInformation() {
+		Package pck  = HnsEsbMainMethod.class.getPackage();
+		final String methodName = LoggingUtil.getMethodName();
+		// init a JSON object
+		JSONObject v2JsonObj = new JSONObject();
+		v2JsonObj.put("Implementation-Version", pck.getImplementationVersion());
```
Comment (weskubo-cgi) [inline hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateVersionInformation.java:31]: I would make "Implementation-Version" a constant.

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

**Pull request merged**